group.go

```diff
@@ -8,26 +8,26 @@ type ErrorGroup interface {
 
 // WalkDeep does a depth-first traversal of all errors.
 // Any ErrorGroup is traversed (after going deep).
-// The visitor function can return false to end the traversal early
-// In that case, WalkDeep will return false, otherwise true
-func WalkDeep(err error, visitor func(err error) bool) bool {
+// The visitor function can return true to end the traversal early
+// In that case, WalkDeep will return true, otherwise false.
+func WalkDeep(err error, visitor func(err error) bool) (done bool) {
```

Comment:
I tend to avoid named parameters since new gophers are not necessarily familiar with them and they create stateful variables. I like the concept of trying to give a name to return types, it would just be nice if they didn't come with the extra ability. If keeping this variable I would also name this "early" rather than "done": from the visitor function perspective it is "done", but from the perspective of calling "WalkDeep", it is always done when it returns.

Reply:
agree~ I rename them.

ps: I am not often use named parameters, too..but preivous days, I read this https://github.com/golang/go/issues/20859 issue make me changed :rofl:

Reply:
Interesting

Reply:
Interesting